django/contrib/admin/options.py

```diff
@@ -373,16 +373,18 @@ def lookup_allowed(self, lookup, value):
         if len(relation_parts) <= 1:
             # Either a local field filter, or no fields at all.
             return True
-        clean_lookup = LOOKUP_SEP.join(relation_parts)
-        valid_lookups = [self.date_hierarchy]
+        valid_lookups = {self.date_hierarchy}
         for filter_item in self.list_filter:
             if isinstance(filter_item, type) and issubclass(filter_item, SimpleListFilter):
-                valid_lookups.append(filter_item.parameter_name)
+                valid_lookups.add(filter_item.parameter_name)
             elif isinstance(filter_item, (list, tuple)):
-                valid_lookups.append(filter_item[0])
+                valid_lookups.add(filter_item[0])
             else:
-                valid_lookups.append(filter_item)
-        return clean_lookup in valid_lookups
+                valid_lookups.add(filter_item)
+
+        # Make sure we also allow lookups which are not accessible locally but allowed in valid_lookups.
+        clean_lookups = {LOOKUP_SEP.join(relation_parts), LOOKUP_SEP.join(relation_parts + [part])}
+        return clean_lookups & valid_lookups
```

Comment:
This returns a set rather than a boolean. Maybe something like  `return not clean_lookups.isdisjoint(valid_lookups)` is better?